test/cql-pytest/test_secondary_index.py

```diff
@@ -64,3 +64,32 @@ def test_partition_order_with_si(cql, test_keyspace):
                 pass
             time.sleep(0.1)
         assert tokens2 == tokens
+
+# Test which ensures that indexes for a query are picked
+# by the order in which they appear in restrictions.
```

Comment:
I would have liked to write this comment to more explicitly explain that there is no real reason why the indexes need to be used in this specific order, but we just decided in issue 7969 that it should be in this order just to allow the user to control the order.
My thinking is that a few years into the future we decide to change this order, and this test fails, and we go to look at this test, we'll immediately be reminded that it didn't really have to be this order - we just decided that it's good that it is - so at that point we will be able to rethink this decision instead of thinking that this test is holy because it is needed for Cassandra compatibility or something.

Reply:
Sure, I'll emphasize that this order may be subject to change in the future, in which case the test case should be remodeled as well. It's actually quite likely to change once we introduce index statistics, based on which we could make a more conscious decision on which index to pick.